frontend/src/utils/routes.ts

```diff
@@ -9,24 +11,42 @@ const ROUTES = {
   SHIPMENT_LIST: '/shipments',
   SHIPMENT_CREATE: '/shipment/new',
   SHIPMENT_VIEW: '/shipment/:shipmentId',
+  SHIPMENT_OFFER_LIST: '/shipment/:shipmentId/offers',
+  SHIPMENT_OFFER_CREATE: '/shipment/:shipmentId/offer/new',
+  SHIPMENT_OFFER_VIEW: '/shipment/:shipmentId/offer/:offerId',
   SHIPMENT_EDIT: '/shipment/:shipmentId/edit',
   KITCHEN_SINK: '/kitchen-sink',
 }
 
 export function groupViewRoute(groupId: number | string) {
-  return ROUTES.GROUP_VIEW.replace(':groupId', groupId.toString())
+  return generatePath(ROUTES.GROUP_VIEW, { groupId })
```

Comment:
I realized that `react-router` has a neat method to generate a path from a route.

```ts
// Example:
generatePath("/shipment/:shipmentId", { shipmentId: "1" })
// -> "/shipment/1"
```

Reply:
ah nice!